lib/ansible/module_utils/powershell/Ansible.ModuleUtils.PrivilegeUtil.psm1

```diff
@@ -0,0 +1,525 @@
+# Copyright (c) 2018 Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+# store in separate variables to make it easier for other module_utils to
+# share this code in their own c# code
+$ansible_privilege_util_namespaces = @(
+    "Microsoft.Win32.SafeHandles",
+    "System",
+    "System.Collections.Generic",
+    "System.Runtime.InteropServices",
+    "System.Security.Principal",
+    "System.Text"
+)
+
+$ansible_privilege_util_code = @'
+namespace Ansible.PrivilegeUtil
```

Comment:
This is really fussy but just wondering if the namespace could simply be 'Ansible.Privilege' - the 'Util' bit of the name doesn't seem to add much to aid understanding?

Reply:
I don't mind what the name should be, I was tossing up whether to copy the name of the module util itself `Ansible.ModuleUtils.PrivilegeUtil` but ended up omitting the ModuleUtils bit to save space. Honestly the only requirement should be that it is unique. Will bring this up on the WWG.

Reply:
After talking to @nitzmahone we ended up keeping the namespace as `Ansible.PrivilegeUtil` to closely reflect the name of the module util itself and to change the class to `Privileges`. The main requirement of the namespace is that it needs to be unique so any common class/object names inside that namespace won't conflict with other module utils we may load.

I don't envision many modules will actually call the C# code itself but will use the PS cmdlets I created below. That doesn't stop them from using it as it exposes some more advanced functions like `DisableAllPrivileges` or the ability to bulk enable/disable/remove a privilege in one call with `SetTokenPrivileges` but that should be a rare situation anyway.